src/Groups/types.jl

```diff
@@ -432,6 +445,10 @@ mutable struct MatrixGroupElem{RE<:RingElem, T<:MatElem{RE}} <: AbstractMatrixGr
    end
 end
 
+function Base.hash(x::MatrixGroupElem, h::UInt)
+  return hash(matrix(x), hash(base_ring(parent(x)), h))
```

Comment:
```suggestion
  return hash(matrix(x), hash(parent(x), h))
```
IMO it is better to compose hash methods, ie just take the hash of the parent here, and then add a hash function for the parent that takes the base_ring into account

Reply:
```suggestion
  return hash(matrix(x), h)
```